File: tests/fixtures/goredis/basic/cache.go
// Fixture for framework-goredis.
package main

import (
	"context"
	"fmt"

	"github.com/redis/go-redis/v9"
)

var rdb = redis.NewClient(&redis.Options{Addr: "localhost:6379"})

func GetUser(ctx context.Context, id int) {
	rdb.Get(ctx, fmt.Sprintf("user:%d", id))
}

func SetUser(ctx context.Context, id int, value string) {
	rdb.Set(ctx, fmt.Sprintf("user:%d", id), value, 0)
}

func IncrementCounter(ctx context.Context) {
	rdb.Incr(ctx, "counter:requests")
}

func DecrementCounter(ctx context.Context) {
	rdb.Decr(ctx, "counter:requests")
}

func HSetUserProfile(ctx context.Context, id int, field, val string) {
	rdb.HSet(ctx, fmt.Sprintf("user:%d:profile", id), field, val)
}

func HGetUserProfile(ctx context.Context, id int) {
	rdb.HGetAll(ctx, fmt.Sprintf("user:%d:profile", id))
}

func AddToSet(ctx context.Context, member string) {
	rdb.SAdd(ctx, "active_users", member)
}

func RemoveFromSet(ctx context.Context, member string) {
	rdb.SRem(ctx, "active_users", member)
}

func AddToSortedSet(ctx context.Context, score float64, member string) {
	rdb.ZAdd(ctx, "leaderboard", &redis.Z{Score: score, Member: member})
}

func QueryLeaderboard(ctx context.Context) {
	rdb.ZRange(ctx, "leaderboard", 0, 10)
}

func LPushQueue(ctx context.Context, item string) {
	rdb.LPush(ctx, "queue:jobs", item)
}

func RPushQueue(ctx context.Context, item string) {
	rdb.RPush(ctx, "queue:jobs", item)
}

func LPopQueue(ctx context.Context) {
	rdb.LPop(ctx, "queue:jobs")
}

func ExpireSession(ctx context.Context, id int) {
	rdb.Expire(ctx, fmt.Sprintf("session:%d", id), 0)
}

func DeleteUser(ctx context.Context, id int) {
	rdb.Del(ctx, fmt.Sprintf("user:%d", id))
}

func PublishEvent(ctx context.Context, channel, message string) {
	rdb.Publish(ctx, channel, message)
}

// ── Dynamic key — non-string arg ──
func DynamicKey(ctx context.Context, key string) {
	rdb.Get(ctx, key)
}

// ── selector binding `s.rdb` inside a struct method ──
type CacheService struct {
	rdb *redis.Client
}

func NewCacheService() *CacheService {
	return &CacheService{rdb: redis.NewClient(&redis.Options{})}
}

func (s *CacheService) Fetch(ctx context.Context, key string) {
	s.rdb.Get(ctx, key)
}

// ── Negative: a method on something that isn't a Redis client ──
type kvStore struct{}

func (k *kvStore) Get(ctx context.Context, key string) string { return key }

func unrelated(ctx context.Context) {
	k := &kvStore{}
	k.Get(ctx, "not_a_redis_key")
}

File: tests/fixtures/redispy/basic/cache.py
"""Fixture for framework-redispy."""

import redis

# Canonical client construction.
rdb = redis.Redis(host='localhost', port=6379)
cache = redis.from_url('redis://localhost:6379/0')


def get_user(uid):
    return rdb.get(f'user:{uid}')


def set_user(uid, data):
    rdb.set(f'user:{uid}', data)


def list_user_keys():
    return rdb.keys('user:*')


def increment_counter():
    rdb.incr('counter:requests')


def decrement_counter():
    rdb.decr('counter:requests')


def hash_set_user(uid, data):
    rdb.hset(f'user:{uid}:profile', mapping=data)


def hash_get_user(uid):
    return rdb.hgetall(f'user:{uid}:profile')


def add_to_set(member):
    rdb.sadd('active_users', member)


def remove_from_set(member):
    rdb.srem('active_users', member)


def add_to_sorted_set(score, member):
    rdb.zadd('leaderboard', {member: score})


def query_leaderboard():
    return rdb.zrange('leaderboard', 0, 10)


def lpush_queue(item):
    rdb.lpush('queue:jobs', item)


def rpush_queue(item):
    rdb.rpush('queue:jobs', item)


def lpop_queue():
    return rdb.lpop('queue:jobs')


def expire_key(uid):
    rdb.expire(f'session:{uid}', 3600)


def delete_user(uid):
    rdb.delete(f'user:{uid}')


# ── publish/subscribe ──
def publish_event(channel, message):
    rdb.publish(channel, message)


# ── alternate client `cache` ──
def cache_get(key):
    return cache.get(key)


def cache_set(key, value):
    cache.set(key, value)


# ── dynamic key — non-string arg ──
def dynamic_key_get(key_var):
    return rdb.get(key_var)


# ── Negative: dict.get/set on something that isn't a Redis client ──
def unrelated():
    d = {'foo': 1}
    return d.get('foo')


# ── self.<client> binding inside a class ──
class CacheService:
    def __init__(self):
        self.r = redis.Redis()

    def fetch(self, k):
        return self.r.get(k)
